share/wake/lib/system/remote_cache_api.wake

```diff
@@ -266,6 +322,9 @@ export def rscApiPostFileBlob (name: String) (file: String) (api: RemoteCacheApi
 #  (RemoteCacheApi "foo" 1 None) | rscApiPostJob (CachePostRequest ...) = Fail "authorization required"
 # ```
 export def rscApiPostJob (req: CachePostRequest) (api: RemoteCacheApi): Result Unit Error =
+    require True = api.getRemoteCacheApiCanPush
+    else failWithError "rsc: RSC Client not configured for pushing to the server"
+
     require Some auth = api.getRemoteCacheApiAuthorization
     else failWithError "rsc: Posting a job requires authorization but none was provided"
```

Comment:
That particular invalid state isn't representable any more (at least not with the old semantics).  I think you should be able to do the same inside `uploadBlobRequest` and avoid having to add the other `require True` checks through the PR.
```suggestion
    require Some auth = api.getRemoteCacheApiAuthorization
    else failWithError "rsc: RSC Client not configured for pushing to the server"
```